### ISSUE TITLE
Assignment 11 - Incorrect ML Preprocessing Procedure

### ISSUE THREAD

**Issue by jerrryyy** [BME1478H/Winter2020class#748]
Hi,

For **Assignment 11**, the .ipynb scales the data before train-test splits (screenshot below). However, this is incorrect; scaling and centering should be done after splitting and only on the training set (`scaler.fit_transform(X_train)`). The parameters derived from the train set should then be applied to the test set (`scaler.transform(X_test)`) to prevent data leakage and biasing the model. The test set should be treated as completely new/unseen data to the model, or else it's no longer generalizable.

![image](https://github.com/BME1478H/Fall2022class/assets/114617121/b2634a94-0b5c-4b45-9d1b-88c35cef90a6)

Also wanted to bring up a super minor nitpick for variable conventions. I believe ML and linear algebra typically keep X uppercase and y lowercase, since X is a matrix, while y is (often) a vector.

Thank you for the fun semester so far,
Jerry

**Comment by cancer-ray** [comment]
yeah that's correct. For anyone else interested, you can read about this data leakage phenomenon from [skl documentation](https://scikit-learn.org/stable/common_pitfalls.html) itself. 

I think we kept it simple for the sake of the  "intro" aspect of the assignment. Learning what ML does, and the concept of training and testing sets is complicated enough for students who touched python for the first time 8 weeks ago.
